werkit/compute/tree/sketch.py

```diff
@@ -0,0 +1,43 @@
+"""
+References:
+
+https://github.com/ivankorobkov/python-inject
+https://github.com/illuin-tech/opyoid
+https://github.com/ets-labs/python-dependency-injector
+"""
+
+
+class MyComputeProcess:
+    a = Input(_type="number")
+    b = Input(_type="number")
+
+    @intermediate(_type="number")
+    def i(self, a):
+        return a
+
+    @intermediate(_type="number")
+    def j(self, b):
+        return b
+
+    @output(_type="number")
+    def r(self, i, j):
+        return i + j
```

Comment:
Per comments below, I would recommend something like this:
```suggestion

class MyComputeProcess:
    a = Input(_type="number")
    b = Input(_type="number")

    @staticmethod
    @intermediate(_type="number")
    def i(self, a):
        return a

    @staticmethod
    @intermediate(_type="number")
    def j(self, b):
        return b

    @staticmethod
    @output(_type="number")
    def r(self, i, j):
        return i + j

```